test/integration/integration-test.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+var _ = require('lodash');
+var assert = require('assert');
+var supertest = require('supertest');
+var wallet = require('./wallet');
+var fixtures = require('../fixtures');
+
+var PREFIX = '/v1';
+var TIMEOUT = 20000;   // how long before each test case times out
+var INTERVAL = 2000;   // how long to wait between checks for validated ledger
+var API = supertest('http://localhost:5990');
+
+
+function verifyTransaction(hash, verify, done) {
+  API.get(PREFIX + '/transactions/' + hash).end(function(error, response) {
+    if (error) {
+      done(error);
+      return;
+    }
+    var body = response.res.body;
+    if (body && body.transaction && body.transaction.validated === true) {
+      verify(body);
+      done(null, response);
+    } else {
+      console.log('NOT VALIDATED YET...');
+      setTimeout(_.partial(verifyTransaction, hash, verify, done), INTERVAL);
+    }
+  });
+}
+
+function testTransaction(url, postData, verify, done) {
+  var isDELETE = (url.indexOf('orders/') !== -1);
+  var method = isDELETE ? API.del : API.post;
+  method(url).send(postData).end(function(error, response) {
+    console.log('PREPARED...');
+    if (error) {
+      done(error);
+      return;
+    }
+    var body = response.res.body;
+    var hash = body.hash;
+    var tx_blob = body.tx_blob;
+    API.post(PREFIX + '/transaction/submit').send({
+      tx_blob: tx_blob
+    }).end(function(_error, _response) {
+      console.log('SUBMITTED...');
+      if (_error) {
+        done(_error);
+        return;
+      }
+      var _body = _response.res.body;
+      assert.strictEqual(_body.engine_result, 'tesSUCCESS');
+      setTimeout(_.partial(verifyTransaction, hash, verify, done), INTERVAL);
+    });
+  });
+}
+
+function setSecret(fixture) {
+  return _.assign({}, fixture, {secret: wallet.getSecret()});
+}
+
+function getURL(type, extra) {
+  var tail = (extra !== undefined ? '/' + extra : '');
+  return PREFIX + '/accounts/' + wallet.getAddress() + '/' + type +
+         tail + '?submit=false';
+}
+
+function verifyResult(transactionType, body) {
+  assert(body && body.transaction);
+  assert.strictEqual(body.transaction.Account, wallet.getAddress());
+  assert.strictEqual(body.transaction.TransactionType, transactionType);
+  assert.strictEqual(body.transaction.meta.TransactionResult, 'tesSUCCESS');
+}
+
+suite.skip('integration tests', function() {
+
+  test('settings', function(done) {
+    this.timeout(TIMEOUT);
```

Comment:
can't you set timeout for the entire suite instead of duplicating for each test?